Add an offset to rootfs image calculation

dwalkes opened the pull request:
Current tegraflash on the warrior branch is failing, hanging when attempting to write the APP partition.  
```
[  11.4855 ] Writing partition APP with core-image-minimal.img
[  11.4866 ] [                                                ] 001%
```
I've found that adding 1M to the partition size is not enough, adding 10M is.  Root cause for mismatch is not yet understood, however the workaround in this branch resolves the issue for me.

moqmar committed:
meta-mender-tegra: add support for jetson-nano development kit

Jetson-nano development kit has a SD-card slot and no emmc. It does not support
nvidia redundant boot feature and therefore there is no dependency to the BUP Payload which
is used on jetson-tx2. Nvidia released a production module of the jetson-nano with 16GB emmc,
but this is not supported by this branch.

Even though there is no emmc, the sd-card is referenced as mmc device in u-boot, which is
why we use ENVIRONMENT_IS_IN_MMC.

Changelog: Title

Signed-off-by: Moritz Marquardt <moritz.marquardt@zeiss.com>
Signed-off-by: Etienne Cordonnier <etienne.cordonnier@taurob.com>

Ecordonnier commented:
I had this issue as long as my image definition contained this line:
IMAGE_ROOTFS_EXTRA_SPACE = "5242880"

After I removed this line it started working again. I had not mentioned it, because I thought it was just a conflict between my rootfs_extra_space config and the partition table introducted with the mender layer (and I can confirm I had the issue before merging the jetson-nano branch).

Commit:
meta-mender-tegra: fix patch not applying cleanly

This fixes this yocto warning:

WARNING: u-boot-tegra-2016.07+gitAUTOINC+8e576d3000-r0 do_patch: Fuzz detected:

Applying patch 0010-tegra-mender-auto-configured-modified.patch
patching file include/config_defaults.h
patching file include/config_distro_bootcmd.h
patching file include/configs/p2771-0000.h
patching file include/configs/tegra-common.h
patching file include/configs/tegra186-common.h
Hunk #1 succeeded at 54 with fuzz 2.

Changelog: None

Signed-off-by: Etienne Cordonnier <etienne.cordonnier@taurob.com>

dwalkes committed:
meta-mender-tegra: Fix image size calculation

See https://github.com/Trellis-Logic/meta-mender-community/pull/8

Fixes support for core-image-minimal and any other images which use a
non-zero value for IMAGE_ROOTFS_EXTRA_SPACE

Changelog: Title
Signed-off-by: Dan Walkes <danwalkes@trellis-logic.com>

dwalkes commented:
> I had this issue as long as my image definition contained this line: IMAGE_ROOTFS_EXTRA_SPACE = "5242880"

Thanks @Ecordonnier the problem was that I didn't support IMAGE_ROOTFS_EXTRA_SPACE in the size calculation.  This was also why `core-image-minimal` didn't work.

I'll put in a PR for this after https://github.com/mendersoftware/meta-mender-community/pull/119 is complete.

mirzak committed:
Merge pull request #119 from taurob/feature/jetson-nano

meta-mender-tegra: add support for jetson-nano development kit
